src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
I think it can fail. If no listening module provides a CScript through signal `ScriptForMining` the miner will continue with a empty script which should result in a unspendable coinbase? But i didn't tested that. Will do.